nav2_mppi_controller/src/critics/obstacles_critic.cpp

```diff
@@ -158,25 +159,21 @@ void ObstaclesCritic::score(CriticData & data)
       if (!near_goal) {
         repulsive_cost[i] += inflation_radius_ - dist_to_obj;
       }
-    }
 
-    if (!trajectory_collide) {all_trajectories_collide = false;}
-    raw_cost[i] = trajectory_collide ? collision_cost_ : traj_cost;
+      if (!trajectory_collide) {all_trajectories_collide = false;}
+      raw_cost(i) = trajectory_collide ? collision_cost_ : traj_cost;
+    }
   }
 
   // Normalize repulsive cost by trajectory length & lowest score to not overweight importance
   // This is a preferential cost, not collision cost, to be tuned relative to desired behaviors
-  auto && repulsive_cost_normalized =
-    (repulsive_cost - xt::amin(repulsive_cost, immediate)) / traj_len;
+  auto repulsive_cost_normalized = (repulsive_cost - repulsive_cost.minCoeff()) / traj_len;
 
   if (power_ > 1u) {
-    data.costs += xt::pow(
-      (critical_weight_ * raw_cost) +
-      (repulsion_weight_ * repulsive_cost_normalized),
-      power_);
+    data.costs +=
+      (critical_weight_ * raw_cost) + (repulsion_weight_ * repulsive_cost_normalized).pow(power_);
```

Comment:
```suggestion
      ((critical_weight_ * raw_cost) + (repulsion_weight_ * repulsive_cost_normalized)).pow(power_);
```
I think this missed an important parenthesis, we should apply pow to both terms